### ISSUE TITLE
torch.fftn gives wrong results on V100

### ISSUE THREAD

**Issue by li-positive-one** [pytorch/pytorch#77048]
### 🐛 Describe the bug

Here is a sample code, notice that all the data on the second axis of B is the same, so I expect that all the data on the second axis of C should be the same as a result of the calculation.

```python
import torch
from torch.fft import fftn,fftshift

#torch.set_default_dtype(torch.double)
device="cuda"
A=torch.rand(1,1,64,32,32)
B=A.expand((1,100,64,32,32)).clone().to(device=device)/64/32/32  # B is same along the second axis

kernel1=torch.rand(64,32,32,30).to(device=device)
kernel2=torch.rand(64,32,32,30).to(device=device)
B=fftshift(B,dim=(-3,-2,-1))

C=0
for i in range(30):
    f1=B*kernel1[:,:,:,i]
    f2=B*kernel2[:,:,:,i]
    f11=fftn(f1,dim=(-3,-2,-1),norm="backward")
    f22=fftn(f2,dim=(-3,-2,-1),norm="backward")
    C=C+f11*f22

print((C-C[:,0:1,...]).abs().max()) # But c is not.
```

However, this is not the case on V100. I tried it on cpu, RTX3090, GTX1080ti, and I got the correct result, i.e. it should output 0. But on V100 it doesn't.

I also tried 
```python
torch.use_deterministic_algorithms(True)
torch.backends.cuda.cufft_plan_cache.max_size = 0
```
but that didn't help.

### Versions

PyTorch version: 1.11.0
Is debug build: False
CUDA used to build PyTorch: 11.3
ROCM used to build PyTorch: N/A

OS: CentOS Linux 7 (Core) (x86_64)
GCC version: (GCC) 11.2.0
Clang version: Could not collect
CMake version: version 3.15.1
Libc version: glibc-2.17

Python version: 3.8.13 (default, Mar 28 2022, 11:38:47)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-3.10.0-1160.11.1.el7.x86_64-x86_64-with-glibc2.17
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: 
GPU 0: Tesla V100-SXM2-16GB
GPU 1: Tesla V100-SXM2-16GB
GPU 2: Tesla V100-SXM2-16GB
GPU 3: Tesla V100-SXM2-16GB

Nvidia driver version: 470.82.01
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.21.5
[pip3] pytorch-lightning==1.4.1
[pip3] torch==1.11.0
[pip3] torchaudio==0.11.0
[pip3] torchmetrics==0.4.1
[pip3] torchvision==0.2.2
[conda] blas                      1.0                         mkl    defaults
[conda] cudatoolkit               11.3.1               h2bc3f7f_2    defaults
[conda] mkl                       2021.4.0           h06a4308_640    defaults
[conda] mkl-service               2.4.0            py38h7f8727e_0    defaults
[conda] mkl_fft                   1.3.1            py38hd3c417c_0    defaults
[conda] mkl_random                1.2.2            py38h51133e4_0    defaults
[conda] numpy                     1.21.5           py38he7a7128_2    defaults
[conda] numpy-base                1.21.5           py38hf524024_2    defaults
[conda] pytorch                   1.11.0          py3.8_cuda11.3_cudnn8.2.0_0    pytorch
[conda] pytorch-lightning         1.4.1              pyhd8ed1ab_0    conda-forge
[conda] pytorch-mutex             1.0                        cuda    pytorch
[conda] torchaudio                0.11.0               py38_cu113    pytorch
[conda] torchmetrics              0.4.1              pyhd8ed1ab_0    conda-forge
[conda] torchvision               0.2.2                      py_3    pytorch

cc @ezyang @gchanan @zou3519 @ngimel @mruberry @peterbell10

**Comment by ezyang** [comment]
there's a decent chance that this is a problem in upstream

**Comment by mruberry** [comment]
What do you think, @peterbell10?

**Comment by peterbell10** [comment]
Unfortunately I don't have a V100 to test on, but I can't see any reason why our kernels would change depending on the device used. So it does seem likely to be a cuFFT bug.

**Comment by mruberry** [comment]
@xwang233 maybe someone at NVIDIA could take a look?

**Comment by xwang233** [comment]
Thanks for the report. I'll take a look.

**Comment by ngimel** [comment]
Can't repro on V100 and cuda 11.5

**Comment by xwang233** [comment]
I can reproduce this with a 1.11 + cuda 11.3 wheel on V100, but can't reproduce it on the latest master + cuda 11.7. Consider this as fixed - unknown?

**Comment by mruberry** [comment]
per @xwang233, looks like this is a bug with CUDA 11.3 and Volta that has since been fixed. We could disable CUDA fftn in that configuration if the results are regularly and significantly incorrect, but if not I think we'll just have to direct users to update their CUDA version to solve the issue. 

Closing for now (unless we decide that the fftn results are so bad we should disable the op on that config).
